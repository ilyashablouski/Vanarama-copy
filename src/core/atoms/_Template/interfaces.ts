import { IBaseProps } from '../../interfaces/base';

export interface ITemplateProps extends IBaseProps {}
