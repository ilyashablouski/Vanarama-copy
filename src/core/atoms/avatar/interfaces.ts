import { IBaseProps } from '../../interfaces/base';
import { TSize } from '../../../types/size';

export interface IAvatarProps extends IBaseProps {
  src?: string;
  size?: TSize;
  altText?: string;
}
