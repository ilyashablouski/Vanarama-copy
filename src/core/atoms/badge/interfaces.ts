import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IBadgeProps extends IBaseProps {
  size?: TSize;
  color?: TColor;
  label?: string;
}
