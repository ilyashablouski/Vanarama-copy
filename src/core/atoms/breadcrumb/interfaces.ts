import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IBreadcrumbProps extends IBaseProps {
  items?: ILink[];
}
