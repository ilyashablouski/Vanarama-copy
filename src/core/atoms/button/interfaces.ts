import { ButtonHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IButtonProps
  extends IBaseProps,
    ButtonHTMLAttributes<HTMLButtonElement> {
  color?: TColor;
  fill?: 'solid' | 'outline' | 'clear';
  icon?: React.ReactNode;
  iconColor?: TColor;
  iconPosition?: 'before' | 'after';
  label?: string | React.ReactNode;
  round?: boolean;
  size?: TSize;
  withoutDefaultClass?: boolean;
  customCTAColor?: string;
}
