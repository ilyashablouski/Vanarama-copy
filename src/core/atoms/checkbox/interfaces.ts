import { InputHTMLAttributes, ReactElement } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ICheckboxProps
  extends IBaseProps,
    InputHTMLAttributes<HTMLInputElement> {
  /**
   * A color for the checkbox.
   */
  color?: TColor;
  /**
   * A unique ID for the checkbox node - this is required here because the
   * underlying checkbox is hidden and therefore you can only click on the label
   * and the label needs to know which input it corresponds to.
   */
  id: string;
  /**
   * 	Invalid state of the checkbox.
   */
  invalid?: boolean;
  /**
   * A name for the input node.
   */
  label: string | (string | ReactElement)[];
  /**
   * Outlined state of the checkbox.
   */
  outline?: boolean;
  /**
   * A ref to attach to the checkbox.
   */
  ref?: React.Ref<HTMLInputElement>;
}
