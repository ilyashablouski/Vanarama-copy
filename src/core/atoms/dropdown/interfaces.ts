import { ReactChild, ReactPortal } from 'react';
import { IBaseProps } from 'core/interfaces/base';

type Children = ReactChild | Array<Children> | ReactPortal;

export type ChildrenCallback = (toggle: () => void) => Children;

export interface IDropdownProps extends IBaseProps {
  label: string;
  defaultOpen?: boolean;
  renderProps?: boolean;
  children?: ChildrenCallback | Children;
}
