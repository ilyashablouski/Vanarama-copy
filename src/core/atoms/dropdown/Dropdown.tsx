import cx from 'classnames';
import React, { useEffect, useRef, useState } from 'react';
import ChevronDownSharp from '../../assets/icons/ChevronDownSharp';
import Icon from '../icon';
import { ChildrenCallback, IDropdownProps } from './interfaces';

const Dropdown = ({
  children,
  className,
  dataTestId,
  defaultOpen = false,
  label,
  renderProps,
}: IDropdownProps) => {
  const [open, setOpen] = useState(defaultOpen);
  const toggle = () => setOpen(prev => !prev);
  const dropdownRef = useRef<null | HTMLDivElement>(null);

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        dropdownRef.current &&
        event.target instanceof Element &&
        !dropdownRef.current.contains(event.target as Element)
      ) {
        setOpen(false);
      }
    };
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [dropdownRef]);

  return (
    <div
      className={cx('dropdown', { '-open': open }, className)}
      data-testid={dataTestId}
      ref={dropdownRef}
    >
      <button className="dropdown--toggle" onClick={toggle} type="button">
        <span>{label}</span>
        <Icon icon={<ChevronDownSharp />} color="dark" />
      </button>
      <div className="dropdown--content">
        {renderProps && children
          ? (children as ChildrenCallback)(toggle)
          : children}
      </div>
    </div>
  );
};

export default Dropdown;
