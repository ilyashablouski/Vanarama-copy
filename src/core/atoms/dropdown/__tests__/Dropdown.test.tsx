import React from 'react';
import renderer from 'react-test-renderer';

import Dropdown from '..';
import { IDropdownProps } from '../interfaces';

const mandatoryProps: IDropdownProps = {
  label: 'Dropdown',
  children: (
    <>
      <span>One</span>
      <span>Two</span>
      <span>Three</span>
    </>
  ),
};

function getComponent() {
  return renderer
    .create(
      <Dropdown label={mandatoryProps.label}>
        {mandatoryProps.children}
      </Dropdown>,
    )
    .toJSON();
}

describe('<Dropdown />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });
});
