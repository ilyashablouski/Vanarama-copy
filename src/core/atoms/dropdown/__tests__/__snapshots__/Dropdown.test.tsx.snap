// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders correctly 1`] = `
<div
  className="dropdown"
>
  <button
    className="dropdown--toggle"
    onClick={[Function]}
    type="button"
  >
    <span>
      Dropdown
    </span>
    <i
      className="icon -dark"
    >
      <svg
        height="1em"
        viewBox="0 0 512 512"
        width="1em"
      >
        <polyline
          points="112 184 256 328 400 184"
          style={
            Object {
              "fill": "none",
              "stroke": "var(--class-color)",
              "strokeLinecap": "square",
              "strokeMiterlimit": 10,
              "strokeWidth": 48,
            }
          }
        />
      </svg>
    </i>
  </button>
  <div
    className="dropdown--content"
  >
    <span>
      One
    </span>
    <span>
      Two
    </span>
    <span>
      Three
    </span>
  </div>
</div>
`;
